lib/db/mem.js

```diff
@@ -1307,13 +1310,19 @@ module.exports = function (log, error) {
 
   Memory.prototype.replaceRecoveryCodes = function (uid, count) {
     uid = uid.toString('hex')
+    let codes
     return getAccountByUid(uid)
       .then(() => {
-        return dbUtil.generateRecoveryCodes(count)
-          .then((codes) => {
-            recoveryCodes[uid] = codes.map((code) => {
+        return dbUtil.generateRecoveryCodes(count, RECOVERY_CODE_KEYSPACE, RECOVERY_CODE_LENGTH)
+          .then((result) => {
```

Comment:
Looks like we can flatten this promise chain to the parent level, so it's all at the same indentation?